pkg/controller/daemon/daemon_controller.go

```diff
@@ -1042,6 +1042,32 @@ func (dsc *DaemonSetsController) simulate(newPod *v1.Pod, node *v1.Node, ds *ext
 		Effect:   v1.TaintEffectNoExecute,
 	})
 
+	// If TaintNodesByCondition feature was enabled, all DaemonSet pods should tolerate
+	// MemoryPressure and DisPressure taints, and the critical pods should tolerate
+	// OutOfDisk taint additional.
+	if utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition) {
```

Comment:
If flag isn't flipped, tainting by nodeCondition wont be applied, so having these tolerations added or not in this case wont change anything right @gmarek ?